src/seals/diagnostics/parabola.py

```diff
@@ -0,0 +1,56 @@
+"""Environment testing for generalization in continuous spaces."""
+
+from gym import spaces
+import numpy as np
+
+from seals import base_envs
+
+
+class ParabolaEnv(base_envs.ResettableMDP):
+    """Environment to mimic parabola curves.
+
+    This environment tests algorithms' ability to learn in continuous
+    action spaces, a challenge for Q-learning methods in particular.
+    The goal is to mimic the path of a parabola p(x) = A*x**2 + B*x +
+    C, where A, B and C are constants sampled uniformly from [-1, 1]
+    at the start of the episode.
+    """
+
+    def __init__(self, x_step: float = 0.05, bounds: float = 5):
+        """Construct environment.
+
+        Args:
+            x_step: x position difference between timesteps.
+            bounds: limits state_space, useful for keeping rewards in
+                a small bounded range.
+        """
+        self._x_step = x_step
+        self._bounds = bounds
+
+        super().__init__(
+            state_space=spaces.Box(low=-bounds, high=bounds, shape=(5,)),
```

Comment:
Bounds seems wrong:
  - The last three components (a, b, and c) are always in `[-1,1]` whatever `bounds` is set to.
  - The first two components (x, y) we say are $\mathbb{R}^2$ in the DERAIL paper. We should either change the paper or the code.

Reply:
The motivation for bounding is more of a technicality than a environment design point: I just wanted to make sure that returns were bounded, (without the bounds, arbitrarily bad policies can get arbitrarily negative returns); I'd rather add that to the paper than change the code.